onnxruntime/core/session/inference_session.cc

```diff
@@ -409,25 +407,20 @@ class InferenceSession::Impl {
       SessionStateInitializer session_initializer{graph, session_state_, execution_providers_,
                                                   kernel_registry_manager_};
 
+      // create SessionState for subgraphs as it's needed by the transformers
+      ORT_RETURN_IF_ERROR(CreateSubgraphSessionState(graph, session_state_));
```

Comment:
I am creating a separate project named optimizer and moving all graph optimization related code to this new project. In the meantime, I am thinking to remove dependency on session_state from graph optimization.

Reply:
Would be great if that can be done.

---
In reply to: [251263710](https://github.com/Microsoft/onnxruntime/pull/368#discussion_r251263710) [](ancestors = 251263710)